portafly/src/components/pages/applications/ApplicationsTable.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect } from 'react'
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  OnSort
+} from '@patternfly/react-table'
+import {
+  PageEmptyState,
+  useDataListFilters,
+  useDataListPagination,
+  useDataListTable,
+  Toolbar,
+  SearchWidget,
+  PaginationWidget,
+  filterRows
+} from 'components'
+import { useTranslation } from 'i18n/useTranslation'
+import { ToolbarContent, ToolbarItem } from '@patternfly/react-core'
+import { IApplication, IPlan } from 'types'
+
+interface Props {
+  applications: IApplication[]
+}
+
+const ApplicationsTable: React.FunctionComponent<Props> = ({ applications }) => {
+  const { t } = useTranslation('applicationsIndex')
+  const {
+    columns,
+    rows,
+    sortBy,
+    setSortBy
+  } = useDataListTable()
+
+  if (rows.length === 0) {
+    return <PageEmptyState msg={t('applications_table.empty_state')} />
+  }
+
+  const { startIdx, endIdx, resetPagination } = useDataListPagination()
+  const { filters } = useDataListFilters()
+
+  // @ts-ignore
+  const planOptions = applications.reduce(
+    (plans, { plan }) => (plans.find((p) => p.id === plan.id) ? plans : [...plans, plan]),
+    [] as IPlan[]
+  ).map((plan) => ({
+    name: String(plan.id),
+    humanName: plan.name
+  }))
```

Comment:
Here we are relying on the plans gathered from the application list, which means there are potentially more plans available but we won't show them in the filter list. Which it might be fine, but @cntlsn and @thomasmaas should be aware of this behaviour

Reply:
I think the intended behavior is to take into account existing plans only better to have it confirmed.

Reply:
I think what is important from a UX standpoint is to only show in the filter list the plans that are actually present in the table at that point in time. How does that sound?

Reply:
Fully agree with @cntlsn, this is about filtering applications by plan(s) so any plan that is not relevant for the collection of applications is irrelevant.

Reply:
I just love commenting stuff that has already been merged